programs/rewards-oracle/src/instructions/set_current_rewards_wrapper_v1.rs

```diff
@@ -0,0 +1,77 @@
+use anchor_lang::prelude::*;
+use helium_entity_manager::state::*;
+use lazy_distributor::program::LazyDistributor;
+use lazy_distributor::state::*;
+use lazy_distributor::{
+  cpi::{accounts::SetCurrentRewardsV0, set_current_rewards_v0},
+  SetCurrentRewardsArgsV0,
+};
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct SetCurrentRewardsWrapperArgsV1 {
+  pub oracle_index: u16,
+  pub current_rewards: u64,
+}
+
+#[derive(Accounts)]
+#[instruction(args: SetCurrentRewardsWrapperArgsV1)]
+pub struct SetCurrentRewardsWrapperV1<'info> {
+  #[account(mut)]
+  // the oracle EOA that gets wrapped
+  pub oracle: Signer<'info>,
+  pub lazy_distributor: Box<Account<'info, LazyDistributorV0>>,
+  #[account(
+    mut,
+    has_one = lazy_distributor
+  )]
+  pub recipient: Box<Account<'info, RecipientV0>>,
+
+  // We assume that the oracle verified that the key to asset entity key went with these rewards.
+  // This endpoint then verifies the connection of that key_to_asset to the recipient asset.
+  #[account(
+    constraint = key_to_asset.asset == recipient.asset,
```

Comment:
You can verify the hash of the entity key to harden against malicious oracles. i.e. pass in `entity_key_hash` in args, verify that `args.entity_key_hash == hash(key_to_asset.entity_key)`

Reply:
We can't really prevent malicious oracles though, they have the key to set rewards for whatever hotspot they want